src/PBE/ExecutableContainer.cs

```diff
@@ -21,100 +21,114 @@ public class ExecutableContainer
+
+            var fsDirs = Directory.EnumerateDirectories(basePath)
+                            .Where(d => Regex.IsMatch(Path.GetFileName(d), pattern, RegexOptions.IgnoreCase))
+                            .Where(d => File.Exists(Path.Combine(d, "FSConsole.exe")));
+
+            if (!fsDirs.Any())
+                return false;
+
+            // Preview Version auslesen
+            if (fsVersion.IsPreview)
+            {
+                fsPath = fsDirs.Where(d => Regex.IsMatch(Path.GetFileName(d), "(.?Preview)$", RegexOptions.IgnoreCase)).FirstOrDefault();
+
+                if (!string.IsNullOrEmpty(fsPath))
+                    return true;
+            }
+
+            // Normale Version auslesen
+            Version maxV = null;
+            foreach (string path in fsDirs.Where(d => Regex.IsMatch(Path.GetFileName(d), @"(?<version>\d\.\d((\.\d*)*)?)$")))
+            {
+                Match m = Regex.Match(Path.GetFileName(path), @"(?<version>\d\.\d((\.\d*)*)?)$");
+
+                if (Version.TryParse(m.Groups["version"].Value, out Version v) && (maxV == null || v > maxV))
                 {
-                    var fsDir = Path.GetFileName(path);
-                    string dirVersionStr = fsDir.Substring(17);
-                    Version dirVersion;
-                    if (Version.TryParse(dirVersionStr, out dirVersion))
-                    {
-                        if (maxVersion == null || dirVersion > maxVersion)
-                        {
-                            maxVersion = dirVersion;
-                            maxPath = path;
-                            maxBeta = null;
-                        }
-                    }
-                    else if (maxVersion == null)
-                    {
-                        // Suche nach Beta-Version
-                        int betaPos = fsDir.IndexOf("Beta");
-                        if (betaPos > 0)
-                        {
-                            int betaVersion = 0;
-                            string betaVersionStr = fsDir.Substring(betaPos + 4);
-                            if (!int.TryParse(betaVersionStr.Trim(), out betaVersion))
-                            {
-                                betaVersion = 0;
-                            }
-                            if (maxBeta == null || betaVersion > maxBeta.Value)
-                            {
-                                maxBeta = betaVersion;
-                                maxPath = path;
-                            }
-                        }
-                    }
+                    maxV = v;
+                    fsPath = path;
                 }
-                if (maxPath != null)
+            }
+
+            if (!string.IsNullOrEmpty(fsPath))
+                return true;
+
+            // Beta Version auslesen
+            int maxBeta = -1;
+            foreach (string path in fsDirs.Where(d => Regex.IsMatch(Path.GetFileName(d), @"(.?Beta(?<beta>\d*))$", RegexOptions.IgnoreCase)))
```

Comment:
Das könnte auch z.B. `4.5.0.0 Beta 7` sein - mit einem Leerzeichen vor der `7`
Den Regex sollten wir deswegen um optionale Whitespaces ergänzen.
Im alten Code wurde `int.TryParse(betaVersionStr.Trim(), ...` ausgeführt.
```suggestion
            foreach (string path in fsDirs.Where(d => Regex.IsMatch(Path.GetFileName(d), @"(.?Beta\s*(?<beta>\d*))$", RegexOptions.IgnoreCase)))
```